src/server/snap.rs

```diff
@@ -50,6 +55,18 @@ pub type Callback = Box<dyn FnOnce(Result<()>) + Send>;
 
 pub const DEFAULT_POOL_SIZE: usize = 4;
 
+// the max duration before a snapshot send task is canceled.
+const SNAP_SEND_TIMEOUT_DURATION: Duration = Duration::from_secs(600);
```

Comment:
What if leader sends a huge snapshot that may takes more than 600s?

Reply:
Ok. I changed get_snap_timeout to also consider snap size. The new timeout will be max(SNAP_SEND_TIMEOUT_DURATION, snap_size / 1MiB * time.Seconds).

Reply:
done.